packages/browser-wallet/src/popup/pages/Setup/PerformRecovery.tsx

```diff
@@ -0,0 +1,103 @@
+import PageHeader from '@popup/shared/PageHeader';
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAtom, useAtomValue } from 'jotai';
+import { useTranslation } from 'react-i18next';
+import { networkConfigurationAtom, seedPhraseAtom } from '@popup/store/settings';
+import { popupMessageHandler } from '@popup/shared/message-handler';
+import { InternalMessageType } from '@concordium/browser-wallet-message-hub';
+import { JsonRpcClient, HttpProvider } from '@concordium/web-sdk';
+import { identitiesAtom, identityProvidersAtom } from '@popup/store/identity';
+import PendingArrows from '@assets/svg/pending-arrows.svg';
+import Button from '@popup/shared/Button';
+import { absoluteRoutes } from '@popup/constants/routes';
+import { BackgroundResponseStatus } from '@shared/utils/types';
+import { getIdentityProviders } from '@popup/shared/utils/wallet-proxy';
+
+// TODO: improve "error state"
+// TODO: Special handling for no identities found
```

Comment:
I assume this is done in the recover identities branch.

Reply:
I added it here :upside_down_face: (A different description for no identities)